site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/service/SiteServiceImpl.java

```diff
@@ -54,6 +55,21 @@ public String save(Site site) {
         if (!validationResponse.isValid()) {
             throw new ValidationException(validationResponse.getErrorMessage());
         }
+
+        validationResponse = validateSiteExists(site.getName());
+        if (!validationResponse.isValid()) {
+            throw new ValidationException(validationResponse.getErrorMessage());
+        }
+
         return siteStore.saveEntity(site);
     }
+
+    private ValidationResponse validateSiteExists(String orgName) {
+        //Check if the Organization already exists.
+        Organization org = siteStore.findOrganizationByName(orgName);
+        if (null != org) {
+            return new ValidationResponse(false, FhirRepo.SITE_EXISTS.message());
```

Comment:
1. Can you move this enum entry to the enum class in this package? Maybe also make all the enum classes package private since that would surface these problems immediately... 
2. You seem to be missing a test for this exception case. Adding it most probably push your code coverage above the required bar by sonar.

Reply:
Addressed as discussed